crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -820,15 +834,24 @@ pub(super) mod _serde {
                     .zip(partition_types)
                     .map(|(v, field)| v.try_into(&field.field_type))
                     .collect::<Result<Vec<_>, _>>()
-            })
-            .transpose()?
-            .unwrap_or_default())
+            } else {
+                Err(Error::new(
+                    crate::ErrorKind::DataInvalid,
```

Comment:
I'm expecting a better error message with more context, for example, the manifest list file path, the entry path, spec id.

Reply:
Have added context of entry path and spec id. We don't have the manifest list file path for now (we pass [u8] to parse directly), I think it need to set in the high level API.